src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobBuilder.cs

```diff
@@ -725,28 +729,45 @@ public int TryWriteBytes(Stream source, int byteCount)
         /// <exception cref="InvalidOperationException">Builder is not writable, it has been linked with another one.</exception>
         public void WriteBytes(ImmutableArray<byte> buffer)
         {
-            WriteBytes(buffer, 0, buffer.IsDefault ? 0 : buffer.Length);
+            if (buffer.IsDefault)
+            {
+                Throw.ArgumentNull(nameof(buffer));
```

Comment:
Why is this being made to throw in a case where it didn't previously throw?

Reply:
It did throw before. If we pass `default(ImmutableArray<byte>)`, the underlying array will be null, and it will throw some lines later.